cocos/core/assets/sprite-frame.ts

```diff
@@ -1019,6 +1158,34 @@ export class SpriteFrame extends Asset {
         }
     }
 
+    public clone ():SpriteFrame {
+        const sp = new SpriteFrame();
+        const v = this.vertices;
+        sp.vertices = v ? {
+            x: v.x,
+            y: v.y,
+            triangles: v.triangles, /* need clone ? */
+            nu: v.nu?.slice(0),
+            u: v.u?.slice(0),
+            nv: v.nv?.slice(0),
+            v: v.v?.slice(0),
+        } : null as any;
+        sp.uv = this.uv.slice(0);
+        sp.uvHash = this.uvHash;
+        sp.unbiasUV = this.unbiasUV.slice(0);
+        sp.uvSliced = this.uvSliced.slice(0);
+        sp._rect = this._rect.clone();
+        sp._offset = this._offset.clone();
+        sp._originalSize = this._originalSize.clone();
```

Comment:
These properties are all created with new SpriteFrame, the clone is overwriting them with new objects, which create garbage objects.
I prefer they are calculated with invocations like `_calculateUV` and `reset` or `sp.rect = this._rect;`

Reply:
These 'garbage objects' are reused in the next change.